compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -876,6 +877,24 @@ DiagnosedSilenceableFailure transform_dialect::IREEBufferizeOp::apply(
     memCpyFn = gpuComprehensiveBufferizeCopyFn;
   }
 
+  //   0. Run enabling transformations.
```

Comment:
Turns out that we do not want to run this during `IREEBufferizeOp`. It should be part of `apply_patterns`, so that we give the `canonicalized_sequence` a chance to clean up afterwards (and before bufferizing).

Reply:
I'm preparing a fix.

Reply:
ok, I'll let you send a new patch, this is just me merging your work anyway :)